### PR TITLE
Add a tool to the b script to help with some yarn tasks

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

File: tools/utility/yarn.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    print(
+        colored(
+            "Cannot find eslint, run `./b yarn` first if you need to format typescript files", "red", attrs=["bold"]
+        )
+    )
+    return None
+
+
+def find_package_json(project_root=".", package_name="NUsight"):
```
Comment (JosephusPaye) [inline tools/utility/yarn.py:23]: This function is probably doing unnecessary work. We can start by just checking that `nusight2/package.json` exists. It's unlikely that it won't be there.

**Comment by KipHamiltons** [comment]
Could you please add information about using this command to NUbook?

It'd probably go in the [build system page](https://nubook.nubots.net/system/foundations/build-system) with the current taxonomy of NUbook, but it seems to me that there's definitely scope for a `b script` page, now that we are expanding it to do more non-build related stuff (tests, and now yarn stuff)

**Comment by JosephusPaye** [comment]
Will do. And yes, it makes since to have a dedicated page for the b scripts now.

File: tools/utility/yarn.py (hunk 1 above)
Comment (Bidski) [reply tools/utility/yarn.py:23]: That is literally the first thing this function does

File: tools/yarn.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+import os
+
+import b
+from dockerise import WrapPty, run_on_docker
```
Comment (Bidski) [inline tools/yarn.py:6]: Since we dont actually run this on docker
```suggestion
from dockerise import WrapPty
```

**Review by Bidski — CHANGES_REQUESTED** [review]

File: tools/utility/yarn.py (hunk 1 above)
Comment (JosephusPaye) [reply tools/utility/yarn.py:23]: You mean after this commit? [`cafae04` (#361)](https://github.com/NUbots/NUbots/pull/361/commits/cafae049a6dc94b80722c4605a218855d843076c) Or before?

Comment (Bidski) [reply tools/utility/yarn.py:23]: Oh, nevermind then :P

**Review by Bidski — DISMISSED** [review]
Looks OK to me

**Comment by KipHamiltons** [comment]
fix the formatting and merge this bad boy @JosephusPaye :sunglasses:

**Review by Bidski — APPROVED** [review]
LGTM

**Pull request merged**